Contributions/CheckList_for_Safety_Claims_on_a_Generic_Linux_System.md

```diff
@@ -0,0 +1,269 @@
+* [Intra-kernel spatial interference](#Intra-kernel-spatial-interference)
+* [Inter-process interference for resources and side effects](#Inter-process-interference-for-resources-and-side-effects)
+
+### **Homogeneity of the test/validation/verification environment**
+
+This list item is not teribly specific to Linux architecture, however it cannot be overlooked, because doing so can compromise the usefulness of any further work.
+
+Field testing and verification are bound to be part of any safety argumentation, because - ultimately - theories must be put to the test, and data must be gathered, to back whatever argumentation might have been put forward.
+
+This cannot happen, until and unless both measurements and verification are aligned with the specific system being analysed. Discrepancy between actual target system and stand-ins can cause different timings and different paths of execution, hiding problems with the target configuration.
+
+Some examples:
+  * **Hardware architecture and revision**
+    * Different architectures (e.g. ARM64 vs x86_64) are not implicitly equivalent, even when it comes to executing generic code, because it is mingled with, and exposed to, architecture specific routines. Furthermore, even if it is the same source code, it will translate to extremely different sets of machine instructions.
+    * Even within the same architecture, there can be various revisions, with different number of registers, peripherals, timing, etc.
+    * Even within the same revision, especially in the case or ARM processors, many hardware features are considered to be optional, or even customer-defined, or configurable. Caches, interrupt routing, presence of certain coprocessors, can (and will) be different, altering the flow of the execution.
+
+  * **Discrete peripherals population**
+
+    Variance in number, type or even make of peripherals can introduce uncontrolled changes in he behaviour of the system, due to difference in device drivers being loaded, event patterns, utilisation of system memory.
+    
+  * **Toolchain parameters**
+
+    Some of the functionality mentioned above might be in different states of activation (certain options are not even binary; e.g. optimisation can have any of 4 values, typically)
+
+  * **Kernel version**
+
+    The fact that the instances of the OS being analysed are all Linux instances is a good start, but far from being sufficient. One needs to confirm that they are the same version, including additional patching that - in the case of distro vendors - might introduce unforeseen variance.
+
+    The truly reliable way to ensure consistency is to compare the SHA signature of the HEAD of the code bases, in case one is attempting to reuse existing measurements from the field. But it's fundamentally a risky approach, and in the signatures are different, then one should proof that the code bases are equivalent, effectively analysing the different patching levels.
```

Comment:
```suggestion
    The truly reliable way to ensure consistency is to compare the SHA signature of the HEAD of the code bases, in case one is attempting to reuse existing measurements from the field. But it's fundamentally a risky approach, and if the signatures are different, then one should prove that the code bases are equivalent, effectively analysing the different patching levels.
```